src/FriendPanel.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useRef } from 'react';
+import { flushSync } from 'react-dom';
+import './FriendPanel.css';
+
+export default function FriendPanel({selectFriend, selectedFriend, message, setMessage, history, setHistory}) {
+    const [friends, setFriends] = useState([]);
+    const [newFriend, setNewFriend] = useState('');
+    const selectedRef = useRef(null);
+  
+    const handleChangeFriend = (event) => {
+      setNewFriend(event.target.value);
+    };
+  
+    function addFriend() {
+      if (friends.includes(newFriend)) {
+        flushSync(() => {selectFriend(newFriend)});
```

Comment:
The app looked fine to me without all the `flushSync`s, so we can probably remove all of them